### DISCUSSION TITLE
information in logfile content

### DISCUSSION THREAD

**Discussion by MingTengCA** [Nek5000/nekRS#542] [Q&A] [answered]
Hello there,

In my logfile, I noticed that there is a ratio at the end of the projP line: 5/8., as shown below.  This ratio goes from 1/8 to 8/8, then starts over all again. Does anyone know what it means? 

--- --- ---
step= 625  t= 3.05544245e+01  dt=5.0e-05  C= 0.50  elapsedStep= 4.24e+00s  elapsedStepSum= 2.72936e+03s
copying solution to nek
calling nek_userchk ...
projP    : resNorm0 7.61e-04  resNorm 3.31e-04  ratio = 2.297e+00  **5/8**
P        : iter 033  resNorm0 3.31e-04  resNorm 9.03e-08
UVW      : iter 001  resNorm0 1.79e-07  resNorm 4.75e-08  divErrNorms 2.08e-07 6.94e-03
step= 626  t= 3.05544741e+01  dt=5.0e-05  C= 0.50  elapsedStep= 3.92e+00s  elapsedStepSum= 2.73328e+03s
copying solution to nek
calling nek_userchk ...
projP    : resNorm0 9.66e-04  resNorm 2.72e-04  ratio = 3.547e+00  **6/8**
P        : iter 040  resNorm0 2.72e-04  resNorm 9.67e-08
UVW      : iter 001  resNorm0 1.77e-07  resNorm 4.51e-08  divErrNorms 2.10e-07 6.93e-03
step= 627  t= 3.05545237e+01  dt=5.0e-05  C= 0.50  elapsedStep= 4.48e+00s  elapsedStepSum= 2.73776e+03s
copying solution to nek
calling nek_userchk ...
projP    : resNorm0 7.90e-04  resNorm 1.75e-04  ratio = 4.500e+00  **7/8**
P        : iter 040  resNorm0 1.75e-04  resNorm 9.07e-08
UVW      : iter 001  resNorm0 1.33e-07  resNorm 6.17e-08  divErrNorms 2.12e-07 6.96e-03
--- --- ---

thanks,
Ming

**Comment by yslan** [comment, marked as answer]
In order to reduce the number of iteration, we use the residual from previous timesteps to build the Krylov subspace and project the current solution out of it. This will give you a better initial guess for linear solver provided the solution is smooth in time and the coefficient of the matrix doesn't change. 

As the solution evolves, such Krylov subspace needs updates in time as well.
The current way is to flush the vectors once it's full. Think about it as the restart GMRES that happens every 8 timesteps.

You can use more vectors by specifying nVector=20, which might reduce your pressure iteration further. 
https://github.com/Nek5000/nekRS/blob/4f87e0e2ec0492e2a3edf27791252d6886814d00/examples/kershaw/kershaw.par#L10
This will however increase your memory usage so you should also monitor the OCCA device memory printed in  the logfile as well.

**Reply by MingTengCA** [reply]
Great! Thank you!
